src/lib/yang/data.lua

```diff
@@ -1428,5 +1428,50 @@ function selftest()
    ]])
    assert(success == false)
 
+   -- Test range / length restrictions.
+   local range_length_schema = schema.load_schema([[module range-length-schema {
+      namespace "urn:ietf:params:xml:ns:yang:range-length-schema";
+      prefix "test";
+
+      leaf-list range_test {
+         type uint8 { range 1..10|20..30; }
+      }
+      leaf-list length_test {
+         type string { length 1..10|20..30; }
+      }
+   }]])
+
+   -- Test range validation. (should fail)
+   local success, err = pcall(load_config_for_schema, range_length_schema,
+                              [[
```

Comment:
Nit: could you move the double bracket to the line above? Since it's just two characters it looks like if there was a blank line between the arguments.

Reply:
Ahh yep - my original patch was against the `lwaftr` branch which has `mem.open_input_string` at the start of the line which I just deleted for this one. Commit updated.

Reply:
That's a good point. Probably is a good idea this patch gets into Snabb via the lwAFTR fork so it's adapted for the latest changes in the library. Although the patch is not strictly lwAFTR related, most of the changes in the YANG library come from Igalia's repository. Let's wait for @wingo opinion before merging it.